ktor-server/ktor-server-host-common/nix/src/io/ktor/server/engine/BaseApplicationResponseNative.kt

```diff
@@ -0,0 +1,312 @@
+/*
+ * Copyright 2014-2021 JetBrains s.r.o and contributors. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package io.ktor.server.engine
+
+import io.ktor.http.*
+import io.ktor.http.content.*
+import io.ktor.server.application.*
+import io.ktor.server.engine.internal.*
+import io.ktor.server.http.*
+import io.ktor.server.plugins.*
+import io.ktor.server.response.*
+import io.ktor.util.*
+import io.ktor.util.cio.*
+import io.ktor.utils.io.*
+import io.ktor.utils.io.concurrent.*
+import kotlinx.coroutines.*
+
+public actual abstract class BaseApplicationResponse actual constructor(
+    actual final override val call: ApplicationCall
+) : ApplicationResponse {
+    private var _status: HttpStatusCode? by shared(null)
+
+    actual override val cookies: ResponseCookies by lazy {
+        ResponseCookies(this, call.request.origin.scheme == "https")
+    }
+
+    actual override fun status(): HttpStatusCode? = _status
+    actual override fun status(value: HttpStatusCode) {
+        _status = value
+        setStatus(value)
+    }
+
+    private var responded by shared(false)
+
+    public actual final override val pipeline: ApplicationSendPipeline = ApplicationSendPipeline(
+        call.application.environment.developmentMode
+    ).apply {
+        merge(call.application.sendPipeline)
+    }
+
+    /**
+     * Commit header values and status and pass them to the underlying engine
+     */
+    protected actual fun commitHeaders(content: OutgoingContent) {
```

Comment:
yep, it's basically the same. I keep the separate version because want to reduce `shared` slowdown for JVM. Will take a look if we can make this part common, but I'm not sure about this

Reply:
Yep, it almost everywhere using `shared` stuff. So basically we can't without replacing every var with shared one on jvm